neon_audio/tts/neon.py

```diff
@@ -0,0 +1,190 @@
+                        }
+                if data not in tts_reqs:
+                    tts_reqs.append(data)
+
+        # General non-server response, use yml configuration
+        else:
+            user_config = get_neon_user_config()["speech"]
+            tts_reqs.append({"speaker": tts_name,
+                             "language": user_config["tts_language"],
+                             "gender": user_config["tts_gender"],
+                             "voice": user_config["neon_voice"]
+                             })
+            if user_config["secondary_tts_language"] and \
+                    user_config["secondary_tts_language"] != user_config["tts_language"]:
+                tts_reqs.append({"speaker": tts_name,
+                                 "language": user_config["secondary_tts_language"],
+                                 "gender": user_config["secondary_tts_gender"],
+                                 "voice": user_config["secondary_neon_voice"]
+                                 })
+    except Exception as x:
+        LOG.error(x)
+
+    LOG.debug(f"Got {len(tts_reqs)} TTS Voice Requests")
+    return tts_reqs
+
+
+class WrappedTTS(TTS):
+    def __new__(cls, base_engine, *args, **kwargs):
+        base_engine.execute = cls.execute
+        base_engine._parse_message = cls._parse_message
+        return cls._init_neon(base_engine, *args, **kwargs)
+
+    @staticmethod
+    def _init_neon(base_engine, *args, **kwargs):
+        """ called after the __init__ method to inject neon-core properties
+        into the selected TTS engine """
+        base_engine = base_engine(*args, **kwargs)
+
+        # TODO ovos-core now also has an internal concept of secondary languages
+        # unify the config !
+        # language_config = Configuration.get()["language"]
+        language_config = get_neon_lang_config()
+
+        base_engine.keys = {}
+
+        base_engine.language_config = language_config
+        base_engine.lang = base_engine.lang or language_config.get("user", "en-us")
+        base_engine.lang_detector = OVOSLangDetectionFactory.create(language_config)
+        base_engine.translator = OVOSLangTranslationFactory.create(language_config)
+
+        # TODO should cache be handled directly in each individual plugin?
+        #   would also allow to do it per engine which can be advantageous
+        neon_cache_dir = NGIConfig("ngi_local_conf").get('dirVars', {}).get('cacheDir')
+        if neon_cache_dir:
+            cache_dir = expanduser(neon_cache_dir)
+            cached_translations = JsonStorage(join(cache_dir, "tx_cache.json"))
+        else:
+            cached_translations = JsonStorageXDG("tx_cache.json", subfolder="neon")
+            cache_dir = dirname(cached_translations.path)
+
+        base_engine.cache_dir = cache_dir
+        base_engine.cached_translations = cached_translations
+        return base_engine
+
+    def _parse_message(self, message, **kwargs):
```

Comment:
Maybe this should be `_get_multiple_tts` or `_handle_request_tts`? Just to better describe what's happening in here